modules/basicfuncs/numeric-funcs.c

```diff
@@ -130,3 +132,156 @@ tf_num_mod(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_num_mod);
+
+
+static GString *
+_make_gstring_buffer(const LogTemplateInvokeArgs *args)
+{
+  if (args->bufs->len == 0)
+    g_ptr_array_add(args->bufs, g_string_sized_new(64));
+
+  return (GString *) g_ptr_array_index(args->bufs, 0);
+}
+
```

Comment:
Yeah, you might be right. If it doesn't lend itself easily I can live with
the private implementation. The whole point of the args member is to
provide a cache instead of having to allocate GString instances on the fast
path. If scratch buffers would be improved (I wouldn't use extend its usage
beyond what it is used for before some serious rethought), then that could
probably be used instead if this private cache.

So if it doesn't fit nicely into template/function.h then let's keep it
private. Perhaps with a better name. It doesn't make GString instances.
On May 3, 2016 3:13 PM, "Tamas Nagy" <notifications@github.com> wrote:

> In modules/basicfuncs/numeric-funcs.c
> <https://github.com/balabit/syslog-ng/pull/1037#discussion_r61877742>:
>
> > @@ -130,3 +132,156 @@ tf_num_mod(LogMessage *msg, gint argc, GString *argv[], GString *result)
> >  }
> >
> >  TEMPLATE_FUNCTION_SIMPLE(tf_num_mod);
> > +
> > +
> > +static GString *
> > +_make_gstring_buffer(const LogTemplateInvokeArgs *args)
> > +{
> > +  if (args->bufs->len == 0)
> > +    g_ptr_array_add(args->bufs, g_string_sized_new(64));
> > +
> > +  return (GString *) g_ptr_array_index(args->bufs, 0);
> > +}
> > +
>
> An alternative refactoring would have been to make it only operate on (GPtrArray
> *) instead. Are you sure this can and will be used at other places as
> well?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/1037/files/ae554114e74e16c1ab7cd3b05a6e6242d0bcbf4a#r61877742>
>